compiler.ts

```diff
@@ -138,13 +139,30 @@ function codeGenExpr(expr: Expr<Type>, env: GlobalEnv): Array<string> {
     case "binop":
       const lhsStmts = codeGenValue(expr.left, env);
       const rhsStmts = codeGenValue(expr.right, env);
-      return [...lhsStmts, ...rhsStmts, codeGenBinOp(expr.op)]
+      if (expr.op == BinOp.Is || expr.op == BinOp.And || expr.op == BinOp.Or) {
+        return [...lhsStmts, ...rhsStmts, codeGenBinOp(expr.op)]
+      }
+      // return [...lhsStmts, ...rhsStmts, codeGenBinOp(expr.op)]
+      var callName = codeGenBinOp(expr.op);
+      return [...lhsStmts, ...rhsStmts, `(call $${callName})`]
 
     case "uniop":
       const exprStmts = codeGenValue(expr.expr, env);
       switch(expr.op){
         case UniOp.Neg:
-          return [`(i32.const 0)`, ...exprStmts, `(i32.sub)`];
+          return [
```

Comment:
Why are the ops above function calls, but this is inline WASM?

Reply:
The binops are function calls to Javascript. To negate a bignum, we invert the sign of the metadata (length). This can be done with simple WASM code.